node/pod.go

```diff
@@ -247,7 +252,17 @@ func (pc *PodController) enqueuePodStatusUpdate(ctx context.Context, q workqueue
 			kpod.lastPodStatusReceivedFromProvider = pod
 			kpod.Unlock()
 			q.AddRateLimited(key)
+			return
+		}
+		// check the pod from lister
+		_, err := pc.podsLister.Pods(pod.Namespace).Get(pod.Name)
+		if err != nil {
+			return
 		}
+		obj := &knownPod{lastPodStatusReceivedFromProvider: pod}
+		pc.knownPods.Store(key, obj)
```

Comment:
This exposes a race if the pod is deleted from API server. If pods.get succeeds, and then the pod is deleted (and we get the deleted callback), we can re-store the pod in known pods, and leak. I think the only "right way" is to add another queue and re-queue.

Reply:
Normally, we would receive an update event before we delete it because we need set the deleteTimeStamp.